### ISSUE TITLE
Share link support

### ISSUE THREAD

**Issue by JeffreyCA** [JeffreyCA/Apollo-ImprovedCustomApi#4]
From https://github.com/EthanArbuckle/Apollo-CustomApiCredentials/issues/36:

> Recently Reddit started implementing a new format for share URLs on their mobile apps. This is not supported by Apollo.
> 
> **how it breaks Apollo** When a user clicks on one of these links, instead of being pushed to the post or comment in question, a WebView opens. This page is often broken if the share link is to another comment, or NSFW post.
> 
> Example:
> 
> 1. shortlinks https://redd.it/1641krf redirects to https://www.reddit.com/r/ModSupport/comments/1641krf/reddit_broke_the_report_forms_by_now_using_a/
> 2. mobile sharelinks https://reddit.com/r/ModSupport/s/9aRRC2rZpd redirects to https://www.reddit.com/r/ModSupport/comments/1641krf/reddit_broke_the_report_forms_by_now_using_a/?share_id=HXzTX3cWsSnrFxIy1IsDg (includes trackable `share_id`)

Current bugs:

- [ ] Share links in private messages still open in the in-app browser
- [ ] Long-tapping share links still pop open a browser page

**Comment by JeffreyCA** [comment]
@losh11, just added support for share links in [v1.0.4](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v1.0.4)! Took quite a bit of trial and error and I might've over-engineered some parts of it but it functions nicely for the most part.

There are currently a few limitations:

- Share links in private messages still open in the in-app browser
- Long-tapping share links still pop open a browser page

Please let me know if you run into any issues with it.

**Comment by JeffreyCA** [comment]
Closing for now, minor bugs remain:

- Share links in private messages still open in the in-app browser
- Long-tapping share links still pop open a browser page

**Comment by Avi0n** [comment]
Ran into this issue today. The following share link opened a webview instead of redirecting to the post.
https://www.reddit.com/r/WutheringWaves/s/0lpkVocMdZ

Tweak version: 1.2.3
Apollo version: 1.15.11

**Comment by Avi0n** [comment]
While trying to gather more info, I became unable to reproduce this issue. But here's what I found before it started working for me:
It seems on my reddit account, it is redirecting to the old reddit site, e.g. `https://old.reddit.com/r/ModSupport/s/9aRRC2rZpd`. This causes a redirection to the Submit Link page: `https://old.reddit.com/r/ModSupport/submit?url=https%3A%2F%2F9aRRC2rZpd`. For my account, I was able to reproduce this even when pasting the old.reddit share URL into a desktop browser.

Unsure if this is related, but when it wasn't working I had somehow signed into reddit in the web view. It seems to have started working around the time I signed out.
